### PR TITLE
RxText and WidgetSelector improvements

### PR DESCRIPTION
**Pull request by pbissonho**
###  Changes
Added ReactiveWidget.
Added errorBuilder and streamEmptyBuilder for RxText and WidgetSelector.
Added tests to ensure that internal behavior remains the same.
###  Breaking Changes
buildEvents renamed to stream in WidgetSelector.

### CODE REVIEW

**Comment by escamoteur** [comment]
Why did you close it?

**Comment by pbissonho** [comment]
> Why did you close it?
To change the destination branch, but I just realized it was just changing the branch right here.
I create a branch called dev to add new changes.

File: lib/src/reactive_widget.dart [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      : assert(stream != null),
+        super(key: key);
+
+  Widget errorBuild(BuildContext context, Object error) {
+    return Center(
+      child: Text(error),
+    );
+  }
+
+  Widget streamEmptyBuild(BuildContext context) {
```
Comment (escamoteur) [inline lib/src/reactive_widget.dart:22]: Wouldn't the name BusyBuilder here be better than empty builder?

File: lib/src/widget_selector.dart [hunk 2]
```diff
@@ -1,170 +1,99 @@
...
-                          }); 
+  Widget build(BuildContext context, data) {
+    if (data) return onTrue(context);
+    return onFalse(context);
   }
-  
-}
-
-
-/// Spinner/Busyindicator that reacts on the output of a `Stream<CommandResult<T>>`. It's made especially to work together with
```
Comment (escamoteur) [inline lib/src/widget_selector.dart:81]: Why did you remove the RxCommandBuilder? I'm pretty sure there a users using it

**Comment by escamoteur** [comment]
@PedroBissonho Sorry for my slow response, we had vacations here

**Comment by escamoteur** [comment]
And overall great Work!!

File: lib/src/reactive_widget.dart (hunk 1 above)
Comment (pbissonho) [reply lib/src/reactive_widget.dart:22]: I didn't think of anything better when I put this name. And the name is really not very cool.
Looking at RxComandBuilder and RxLoader, when the stream has no data and no error the return is placeHolderBuilder.

```Dart
if (!lastReceivedItem.hasData && !lastReceivedItem.hasError) {
      if (widget.placeHolderBuilder != null) {
        return widget.placeHolderBuilder(context);
      }
    }
```

Since the behavior is the same, wouldn't it be better to rename it to placeHolderBuilder?

File: lib/src/widget_selector.dart (hunk 2 above)
Comment (pbissonho) [reply lib/src/widget_selector.dart:81]: I just moved to the rx_command_builder.dart file.

Comment (escamoteur) [reply lib/src/widget_selector.dart:81]: ah, I'm just wondering why the new file isn't listed when doing a review

File: lib/src/reactive_widget.dart (hunk 1 above)
Comment (escamoteur) [reply lib/src/reactive_widget.dart:22]: Sorry I somehow thought I had seen it used in case of a waiting StreamBuilder. I think `placeHolderBuilder` is a better solution. I saw that you named them `Build`and not `Builder`any reason for that?

**Comment by escamoteur** [comment]
For my understanding, why did you introduce the `ReactiveBuilderBase`  it's only used in `ReactiveBuilder`

**Comment by PedroBissonho** [comment]
> For my understanding, why did you introduce the `ReactiveBuilderBase` it's only used in `ReactiveBuilder`
I added this way because in the other lib I was using `ReactiveBuilderBase for other widgets.
It really is not making much sense and maybe so many Inheritance don't be so good.
The`ReactiveBuilderBase` removed on new pull request.

**Comment by pbissonho** [comment]
@escamoteur 

> For my understanding, why did you introduce the `ReactiveBuilderBase` it's only used in `ReactiveBuilder`

I added this way because in the other lib I was using `ReactiveBuilderBase for other widgets.
It really is not making much sense and maybe so many Inheritance don't be so good.
The`ReactiveBuilderBase` removed on new pull request.

File: lib/src/reactive_widget.dart (hunk 1 above)
Comment (pbissonho) [reply lib/src/reactive_widget.dart:22]: I put it as 'Build' instead of 'Builder' to follow the 'build' method.
And also to differentiate when it is the internal method or when it receives as a parameter.
What do you think about ? Is it good that way?

``` dart
// when it comes to the internal method.
Widget build(BuildContext context, T data);
Widget errorBuild(BuildContext context, Object error);
Widget placeHolderBuild(BuildContext context);
``` 
``` dart
// When it is received as a parameter.
final RxBuilder<T> builder;
final ErrorBuilder<T> errorBuilder;
final PlaceHolderBuilder placeHolderBuilder;
```
